### ISSUE TITLE
AttributeError: 'Registry' object has no attribute 'keys'

### ISSUE THREAD

**Issue by Pixelartist** [airbnb/knowledge-repo#452]
Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

I need you assistance with an error I am receiving. I am using the latest knowledge base build on debian with postgres. After a dev server change, I am trying to reinitate the service but the frontend returns me the mentioned error (see below for full error log).

```
INFO:werkzeug:37.44.7.154 - - [24/Sep/2018 13:26:24] "GET /feed HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.5/dist-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.5/dist-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/models.py", line 194, in __call__
    raise_with_traceback(e)
  File "/usr/local/lib/python3.5/dist-packages/future/utils/__init__.py", line 419, in raise_with_traceback
    raise exc.with_traceback(traceback)
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/models.py", line 187, in __call__
    return self._route(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/flask_principal.py", line 199, in _decorated
    rv = f(*args, **kw)
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/routes/index.py", line 88, in render_feed
    post.tldr = render_post_tldr(post)
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/utils/render.py", line 31, in render_post_tldr
    return markdown.Markdown(extensions=MARKDOWN_EXTENSTIONS).convert(post.tldr.strip())
  File "/usr/local/lib/python3.5/dist-packages/markdown/core.py", line 100, in __init__
    configs=kwargs.get('extension_configs', {}))
  File "/usr/local/lib/python3.5/dist-packages/markdown/core.py", line 128, in registerExtensions
    ext._extendMarkdown(self)
  File "/usr/local/lib/python3.5/dist-packages/markdown/extensions/__init__.py", line 82, in _extendMarkdown
    self.extendMarkdown(md, {})
  File "/usr/local/lib/python3.5/dist-packages/markdown/extensions/headerid.py", line 85, in extendMarkdown
    if 'attr_list' in md.treeprocessors.keys():
AttributeError: 'Registry' object has no attribute 'keys'
```

Im not sure where to start with this one. I created a new rep from scratch which is working. Any help appreciated.

**Comment by matthewwardrop** [comment]
Hi @Pixelartist ; this looks to be a `markdown` related issue. Do you know which version you have installed?

This should be discoverable using: `markdown.__version__.version`

**Comment by matthewwardrop** [comment]
@Pixelartist It looks like `markdown` v3.0 has just been released, which probably is the cause of this problem. Try running `pip install markdown<3` and everything will work I expect. We'll have to take a deeper look at what's going on here soon, or it will start biting more people!

**Comment by Pixelartist** [comment]
Hello @matthewwardrop ,

you are correct. I rebuild the server and replaced the markdown with a lower version and am back in business again. I will leave this bug open for you guys to find out what changed in version 3.

For everyone else who encounters this, uninstall the latest version and add a lower version of the markdown by using:

```
pip uninstall markdown
pip install 'markdown<3'
```

This should install version 2.6.11 of markdown.

**Comment by babybear1992** [comment]
Hi @matthewwardrop 
`markdown 2.6.11` works totally fine, thanks a lot.

**Comment by matthewwardrop** [comment]
This is addressed in #453 (thanks @naoyak). Once this is cleaned up and merged, I'll release a new version to solve this issue for future users.

**Comment by brijeshgzp05** [comment]
yes you are correct. Version here iis the problem.
